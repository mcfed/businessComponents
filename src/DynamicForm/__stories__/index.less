.testClass {
  .ant-form-item {
    width: 50%;
    // float: left;
    display: inline-block;
  }
}
