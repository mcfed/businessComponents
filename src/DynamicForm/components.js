import React from 'react';
import {Input, Select, Checkbox, Radio} from 'antd';

const Option = Select.Option;
const CheckboxGroup = Checkbox.Group;
const RadioGroup = Radio.Group;

export class InputForm extends React.Component {
  render() {
    const {} = this.props;
    return <Input {...this.props} />;
  }
}

export class SelectForm extends React.Component {
  render() {
    const {} = this.props;
    return <Select {...this.props} />;
  }
}

export class CheckboxForm extends React.Component {
  render() {
    const {} = this.props;
    return <Checkbox {...this.props} />;
  }
}

export class CheckboxGroupForm extends React.Component {
  render() {
    const {} = this.props;
    return <CheckboxGroup {...this.props} />;
  }
}

export class RadioGroupForm extends React.Component {
  render() {
    const {} = this.props;
    return <RadioGroup {...this.props} />;
  }
}
