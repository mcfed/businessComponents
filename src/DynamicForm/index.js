import React from 'react';
import PropTypes from 'prop-types';
import FormItem from '../FormItem';
import BaseForm from '../BaseForm';

export default class DynamicForm extends React.Component {
  saveFormRef(form) {
    this.form = form;
  }
  render() {
    const {formData, className, ...otherProp} = this.props;
    return (
      <BaseForm
        {...otherProp}
        className={className}
        ref={this.saveFormRef.bind(this)}>
        {formData.map(item => {
          const {component, ...others} = item;
          return React.createElement(
            FormItem,
            {key: others.name, name: others.name, children: others.children},
            React.createElement(component, others)
          );
        })}
      </BaseForm>
    );
  }
}

DynamicForm.propTypes = {
  /**
  tab 配置
  **/
  formData: PropTypes.array
};
