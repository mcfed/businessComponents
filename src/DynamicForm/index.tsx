import React from 'react';
import FormItem from '../FormItem';
import BaseForm from '../BaseForm';

/**
 * formData：步骤信息数组，必须
 * className: 样式类名，非必须
 */
interface DynamicFormProps {
  /**
   * 步骤信息数组，必须
   */
  formData: object[];
  /**
   * 样式类名，非必须
   */
  className?: string;
}

interface State {}

export default class DynamicForm extends React.Component<
  DynamicFormProps,
  State
> {
  form: any;
  saveFormRef(form: any) {
    this.form = form;
  }
  render() {
    const {formData, className, ...otherProp} = this.props;
    return (
      <BaseForm
        {...otherProp}
        // @ts-ignore
        className={className}
        ref={this.saveFormRef.bind(this)}>
        {formData.map((item: any) => {
          const {component, ...others} = item;
          return React.createElement(
            FormItem,
            {key: others.name, name: others.name, children: others.children},
            React.createElement(component, others)
          );
        })}
      </BaseForm>
    );
  }
}
