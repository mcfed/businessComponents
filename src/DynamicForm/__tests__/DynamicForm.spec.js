import {shallow} from 'enzyme';
import React from 'react';
import {Input} from 'antd';

import DynamicForm from '../index.tsx';

const setup = props => {
  const wrapper = shallow(<DynamicForm {...props} />);

  return {
    wrapper,
    props
  };
};

describe('快照测试', () => {
  it('全页快照', () => {
    const props = {
      formData: [
        {
          name: 'input1',
          component: Input,
          label: 'input1',
          rules: [
            {
              required: true,
              message: '必填input1'
            }
          ],
          defaultValue: 'input1',
          formItemLayout: {
            labelCol: {span: 4},
            wrapperCol: {span: 8}
          }
        }
      ]
    };
    const {wrapper} = setup(props);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('全方法测试', () => {
  const props = {
    formData: [
      {
        name: 'input1',
        component: Input,
        label: 'input1',
        rules: [
          {
            required: true,
            message: '必填input1'
          }
        ],
        defaultValue: 'input1',
        formItemLayout: {
          labelCol: {span: 4},
          wrapperCol: {span: 8}
        }
      }
    ]
  };
  it('saveFormRef 方法测试', () => {
    const {wrapper} = setup(props);
    const instance = wrapper.instance();
    instance.saveFormRef('123');
    expect(instance.form).toBe('123');
  });
});
