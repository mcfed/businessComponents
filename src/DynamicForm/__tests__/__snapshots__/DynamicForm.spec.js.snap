// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 全页快照 1`] = `
<Form(BaseForm)>
  <FormItemRender
    key="input1"
    name="input1"
  >
    <Input
      defaultValue="input1"
      formItemLayout={
        Object {
          "labelCol": Object {
            "span": 4,
          },
          "wrapperCol": Object {
            "span": 8,
          },
        }
      }
      label="input1"
      name="input1"
      rules={
        Array [
          Object {
            "message": "必填input1",
            "required": true,
          },
        ]
      }
      type="text"
    />
  </FormItemRender>
</Form(BaseForm)>
`;
